### PR TITLE
Remove the old hierarchy of AbstractStreamFileAdmin

### PR DESCRIPTION
**Pull request by chtyim**
It was only needed for 2.2->2.3 transition.

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java [hunk 1]
```diff
@@ -247,7 +226,7 @@ public void updateConfig(StreamConfig config) throws IOException {
       // This call will also save the config
       streamCoordinatorClient.changeTTL(originalConfig, config.getTTL());
     }
-    if (originalConfig.getNotificationThresholdMB() != config.getNotificationThresholdMB()) {
+    if (!originalConfig.getNotificationThresholdMB().equals(config.getNotificationThresholdMB())) {
```
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java]: this change is just to do the right comparison. There will be proper fix for the update config issue after this PR. This PR is a stepping stone to first simplify the class hierarchy before making more changes.

**Comment by anew** [comment]
always nice to get rid of some legacy code. :)
:+1:

**Comment by anew** [comment]
How did that unit test even build? Looks good to me now.

**Comment by chtyim** [comment]
Both implementation shares the same interface, but the unit test has certain assumptions on the implementation

**Comment by chtyim** [comment]
Build passed: https://builds.cask.co/browse/CDAP-DUT700-2

**Pull request merged**